### PR TITLE
Ra dspdc 837 map biosamples

### PR DESCRIPTION
**Pull request by raaidbroad**
Attempt at the biosample mapping, barring 5 fields (which have TODO comments above them).

### CODE REVIEW

File: transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/BiosampleTransformations.scala [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  def transformBiosample(biosampleInput: Msg): Biosample = {
+    val (auditLevel, auditLabels) = CommonTransformations.summarizeAudits(biosampleInput)
+    Biosample(
+      id = CommonTransformations.readId(biosampleInput),
+      crossReferences = biosampleInput.read[Array[String]]("dbxrefs"),
+      timeCreated = biosampleInput.read[OffsetDateTime]("date_created"),
+      source = biosampleInput.tryRead[String]("source"),
+      dateObtained = biosampleInput.tryRead[LocalDate]("date_obtained"),
+      derivedFromBiosampleId =
+        biosampleInput.tryRead[String]("library").map(CommonTransformations.transformId),
```
[the remark this answers is not in the document]
Comment (raaidbroad) [reply transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/BiosampleTransformations.scala:22]: whoops bad copy/paste

File: transformation/src/test/scala/org/broadinstitute/monster/encode/transformation/TransformationPipelineBuilderSpec.scala [hunk 2]
```diff
@@ -17,10 +17,9 @@ class TransformationPipelineBuilderSpec extends PipelineBuilderSpec[Args] {
   override val testArgs =
     Args(inputPrefix = inputDirString, outputPrefix = compareDirString)
 
-  override def afterAll(): Unit = {
+  override def afterAll(): Unit =
     compareDir.delete()
-    ()
-  }
+  ()
```
[the remark this answers is not in the document]
Comment (raaidbroad) [reply transformation/src/test/scala/org/broadinstitute/monster/encode/transformation/TransformationPipelineBuilderSpec.scala:22]: sure did hahah

File: transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/BiosampleTransformations.scala [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      dateObtained = biosampleInput.tryRead[LocalDate]("date_obtained"),
+      derivedFromBiosampleId =
+        biosampleInput.tryRead[String]("part_of").map(CommonTransformations.transformId),
+      // TODO needs biosampleType join
+      anatomicalSite = "site",
+      // TODO needs biosampleType join
+      biosampleType = "type",
+      samplePreservationState = biosampleInput.tryRead[String]("preservation_method"),
+      seeAlso = biosampleInput.tryRead[String]("url"),
+      donorId = biosampleInput.read[String]("donor"),
```
Comment (danxmoran) [inline transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/BiosampleTransformations.scala:29]: This also needs a `CommonTransformations.transformId` call

File: transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/BiosampleTransformations.scala [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      auditLabels = auditLabels,
+      maxAuditFlag = auditLevel,
+      award = biosampleInput.read[String]("award"),
+      // TODO needs replicates
+      biologicalReplicateId = 42,
+      cellIsolationMethod = biosampleInput.tryRead[String]("cell_isolation_method"),
+      geneticModifications = biosampleInput.read[Array[String]]("applied_modifications"),
```
Comment (danxmoran) [inline transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/BiosampleTransformations.scala:36]: Are we sure this array will always be present in the source data?

File: transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/BiosampleTransformations.scala [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+      healthStatus = biosampleInput.tryRead[String]("health_status"),
+      lab = biosampleInput.read[String]("lab"),
+      // TODO needs to join with experiments
+      replicationType = "repType",
+      // TODO needs replicates
+      technicalReplicateId = 43,
+      treatments = biosampleInput.read[Array[String]]("treatments"),
```
Comment (danxmoran) [inline transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/BiosampleTransformations.scala:43]: Same with this array

File: transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/BiosampleTransformations.scala [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+      wasPerturbed = biosampleInput.read[Boolean]("perturbed"),
```
Comment (danxmoran) [inline transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/BiosampleTransformations.scala:44]: Are we sure this boolean will always be present in the source data?

File: transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/BiosampleTransformations.scala (hunk 4 above)
Comment (raaidbroad) [reply transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/BiosampleTransformations.scala:36]: Not sure; on a quick glance it seems like it is, but the array can be empty

File: transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/BiosampleTransformations.scala (hunk 5 above)
Comment (raaidbroad) [reply transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/BiosampleTransformations.scala:43]: Not sure; on a quick glance it seems like it is, but the array can be empty, lots of empty arrays

File: transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/BiosampleTransformations.scala (hunk 6 above)
Comment (raaidbroad) [reply transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/BiosampleTransformations.scala:44]: Same as the arrays, it seems like it is present everywhere but not entirely sure

**Pull request merged**